### PR TITLE
Add Custom Alerts and Toasts

### PR DESCRIPTION
**Pull request by NathHorrigan**
This PR adds custom UI for alerts and toasts (like a warning) that can be used across the app. The benefit of custom alerts over native ones are:

- You can stack multiple alerts on iOS and Android
- You can store a ref to programmatically dismiss the alert

Screenshots:
IMG_3626
IMG_3625

### CODE REVIEW

File: src/containers/InstructionsContainer.tsx [hunk 1]
```diff
@@ -140,6 +154,15 @@ export const InstructionsContainer: ExperimentModule<InstructionsModuleState> =
   const CurrentScreen = IntroScreen ?? TrialScreen
   const inRatingPhase = IntroScreen === undefined && TrialScreen !== undefined
 
+  // Generate a componennt with appropariate key to optimize rendering
```
Comment (mixxorz) [inline src/containers/InstructionsContainer.tsx]: Typo: component

File: src/redux/reducers/alerts.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+      return state
+    })
+
+    builder.addCase(dismissAlert, (state, action) => {
+      // Remove any alerts with id
+      state.alerts = state.alerts.filter((alert) => {
+        if (alert.id != action.payload) return true
+
+        return false
```
Comment (mixxorz) [inline src/redux/reducers/alerts.ts]: This can be shortened to `return alert.id !== action.payload`.

File: src/redux/reducers/alerts.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+      return state
+    })
+
+    builder.addCase(dismissToast, (state, action) => {
+      // Remove any alerts with id
+      state.toasts = state.toasts.filter((toast) => {
+        if (toast.id != action.payload) return true
+
+        return false
```
Comment (mixxorz) [inline src/redux/reducers/alerts.ts]: Ditto for this one.

**Review by mixxorz — APPROVED** [review]
Looks good to me! Thanks Nathan.

File: src/redux/reducers/alerts.ts (hunk 2 above)
Comment (NathHorrigan) [reply src/redux/reducers/alerts.ts]: Of course! Thanks 😂

**Pull request merged**